src/sentry/http.py

```diff
@@ -183,3 +221,139 @@ def safe_urlopen(url, method=None, params=None, data=None, json=None,
 
 def safe_urlread(response):
     return response.content
+
+
+def expose_url(url):
+    if url is None:
+        return u'<unknown>'
+    if url[:5] == 'data:':
+        return u'<data url>'
+    url = truncatechars(url, MAX_URL_LENGTH)
+    if isinstance(url, six.binary_type):
+        url = url.decode('utf-8', 'replace')
+    return url
+
+
+def fetch_file(url, headers=None, domain_lock_enabled=True, outfile=None):
+    """
+    Pull down a URL, returning a UrlResult object.
+    """
+    # lock down domains that are problematic
+    if domain_lock_enabled:
+        domain = urlparse(url).netloc
+        domain_key = 'source:blacklist:v2:%s' % (
+            md5_text(domain).hexdigest(),
+        )
+        domain_result = cache.get(domain_key)
+        if domain_result:
+            domain_result['url'] = url
+            raise CannotFetch(domain_result)
+
+    logger.debug('Fetching %r from the internet', url)
+
+    http_session = build_session()
+    response = None
+    try:
+        try:
+            start = time.time()
+            response = http_session.get(
+                url,
+                allow_redirects=True,
+                verify=False,
+                headers=headers,
+                timeout=settings.SENTRY_FETCH_SOCKET_TIMEOUT,
+                stream=True,
+            )
+
+            try:
+                cl = int(response.headers['content-length'])
+            except (LookupError, ValueError):
+                cl = 0
+            if cl > settings.SENTRY_FETCH_MAX_SIZE:
+                raise OverflowError()
+
+            contents = []
```

Comment:
Is it significant enough that it matters here? Even with `cStringIO`? I assume writing doesn't incur `O(N*M)` performance like appending to a string at least. My thinking is that a API to match would outweigh a slight performance penalty.